solr/core/src/java/org/apache/solr/response/SolrQueryResponse.java

```diff
@@ -139,30 +142,63 @@ public ReturnFields getReturnFields() {
 
   /**
    * If {@link #getResponseHeader()} is available, set {@link #RESPONSE_HEADER_PARTIAL_RESULTS_KEY}
-   * flag to true.
+   * attribute to true or "omitted" as required.
    */
-  public void setPartialResults() {
+  public void setPartialResults(SolrQueryRequest req) {
     NamedList<Object> header = getResponseHeader();
-    if (header != null
-        && header.get(SolrQueryResponse.RESPONSE_HEADER_PARTIAL_RESULTS_KEY) == null) {
-      header.add(SolrQueryResponse.RESPONSE_HEADER_PARTIAL_RESULTS_KEY, Boolean.TRUE);
+    if (header != null) {
+      if (header.get(SolrQueryResponse.RESPONSE_HEADER_PARTIAL_RESULTS_KEY) == null) {
+        Object value = partialResultsStatus(shouldDiscardPartials(req.getParams()));
+        header.add(SolrQueryResponse.RESPONSE_HEADER_PARTIAL_RESULTS_KEY, value);
+      }
     }
   }
 
+  public static Object partialResultsStatus(boolean discarding) {
+    return discarding ? "omitted" : Boolean.TRUE;
```

Comment:
Should this be a constant somewhere?

Reply:
It could be. Though it only gets used here and in one assert in one test right now.